client/signup/jetpack-connect/authorize-form.jsx

```diff
@@ -521,13 +511,35 @@ const LoggedInForm = React.createClass( {
 	}
 } );
 
-const JetpackConnectAuthorizeForm = React.createClass( {
-	displayName: 'JetpackConnectAuthorizeForm',
-	mixins: [ observe( 'userModule' ) ],
+class JetpackConnectAuthorizeForm extends Component {
+	static propTypes = {
+		authAttempts: PropTypes.number,
+		authorize: PropTypes.func,
+		calypsoStartedConnection: PropTypes.bool,
+		createAccount: PropTypes.func,
+		goBackToWpAdmin: PropTypes.func,
+		goToXmlrpcErrorFallbackUrl: PropTypes.func,
+		isAlreadyOnSitesList: PropTypes.bool,
+		isFetchingSites: PropTypes.bool,
+		jetpackConnectAuthorize: PropTypes.shape( {
+			queryObject: PropTypes.shape( {
```

Comment:
If you're to list all that `queryObject` can support, you can definitely add some more of those, for example:

* `already_authorized`
* `new_user_started_connection`
* `redirect_after_auth`
* `site`
* `jp_version`

I haven't done a good audit of what props can be passed here, but the list already looks a little scary 😱